### DISCUSSION TITLE
Using `npm publish` publishes only part of the files

### DISCUSSION THREAD

**Discussion by YonatanKra** [community/community#26808] [Actions]
<p>In one of my repositories I’ve started publishing the files using gha.<br>
In the same repo, when doing <code>npm publish</code>, the package would have publish as expected.<br>
When I’m using the same command in the repository, only some of the files are being published.<br>
In my <code>package.json</code> I’ve set npm to publish two folders:</p>
<pre><code class="lang-auto">  "files": [
    "dist/**/*",
    "template/**/*"
  ],
</code></pre>
<p>Here are images from the repository’s relevant action:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/1/116e3bdee48d7b9cb014102470df35465bd3a6c3.png" data-download-href="/uploads/short-url/2uci6lHnDB3SsRna6KmGfisSoo3.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/6459899/181099982-d7158a90-7b26-4ea7-a14b-dd8c356ecf38.png" alt="image" data-base62-sha1="2uci6lHnDB3SsRna6KmGfisSoo3" width="326" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/1/116e3bdee48d7b9cb014102470df35465bd3a6c3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">406×621 40.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
You can see that the <code>template</code> folder exists but only a few files are packaged. The template folder has many more files and folders.
<p>What can be the problem here? Is there anything different when publishing from gha?</p>
<p>Thanks!</p>

**Comment by YonatanKra** [comment]
<p>When using <code>ls</code>, I can see the files are already there:<br>
<img src="https://user-images.githubusercontent.com/6459899/181099983-91f7b3d5-757f-4293-b1dc-04b9ab7a9fdc.png" alt="image" data-base62-sha1="27NjYu1YZuPkxtn17V8AzeoEWNV" width="260" height="386"></p>

**Comment by YonatanKra** [comment]
<p>When I publish locally (on my machine) the publish works as expected:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/4/347a1b7c783c4c5ae6ed6d6833662a96070f2e64.png" data-download-href="/uploads/short-url/7uerqaKlLwfXCgFuK8XVzZ3hnYU.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/6459899/181099985-1320e0e2-0c7f-4665-9122-8ec737799c09.png" alt="image" data-base62-sha1="7uerqaKlLwfXCgFuK8XVzZ3hnYU" width="353" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/4/347a1b7c783c4c5ae6ed6d6833662a96070f2e64_2_353x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/4/347a1b7c783c4c5ae6ed6d6833662a96070f2e64_2_529x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/3/4/347a1b7c783c4c5ae6ed6d6833662a96070f2e64_2_706x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/4/347a1b7c783c4c5ae6ed6d6833662a96070f2e64_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">847×1198 162 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by YonatanKra** [comment, marked as answer]
<p>Found the issue - it was the npm version.<br>
gha has version 6.x, while locally I’ve been using 7.x.</p>

**Comment by lirantal** [comment]
<p>Thanks for the update <a class="mention" href="/yonatankra">@YonatanKra</a>.<br>
Did you figure out what was the actual issue for which npm6 didn’t publish as you expected?</p>

**Comment by YonatanKra** [comment]
<p>Yes. I had a <code>package.json</code> file in one of my templates. It was taken as a “real” package.json file and interfered with the actual deployment to npm.<br>
Note to self - if you add <code>tmpl</code> suffix to <code>html</code> and <code>js</code> files - do the same for <code>json</code> files cause you can never know :slight_smile:</p>

**Comment by lirantal** [comment]
<p>Ahh gotcha! Glad you caught up to that nasty one :slight_smile:</p>
